### PR TITLE
Third person view

### PR DESCRIPTION
**Pull request by meszaros-lajos-gyorgy**
I'm experimenting with adding third person view to the game: https://youtu.be/62hbFL8iZ1I
It's very far from being complete, but I thought it would be nice to share my progress so far, so that we can start a discussion about it.

### CODE REVIEW

**Comment by dscharrer** [comment]
Instead of ARX_INTERACTIVE_Show_Hide_1st() we should probably set EXTERNALVIEW to false in third person mode so that the player character is fully rendered like when a scripted camera is active, but need to check all uses - e.g. maybe the player character should still be rendered in front of everything even in 3rd person.

It is actually possible to implement a third-person or top down camera in AL 1.2+ using scripts - I played around with that a while ago:
 https://arx-libertatis.org/files/mods/doggo.zip
Use the dog to switch camera modes - this also affects the death camera. Feed the dog for some extra fun.

A real implementation in the engine still makes sense since there we can handle cases where the camera would clip into level geometry and either move it closer to the player or hide some of that geometry.

**Comment by meszaros-lajos-gyorgy** [comment]
Doesn't `EXTERNALVIEW` get resetted every time the game loads a new level or a cutscene ends? I was also thinking about attaching a camera entity to the player, which would follow him, but I only found partial support for detecting various movements via events. For example there is `on move`, but there is no `on crouch` or `on jump` (don't know if on move detects vertical movement too)

**Comment by blackerking** [comment]
It sounds like a camera used in Gothic / Risen.

File: src/core/ArxGame.cpp [hunk 1]
```diff
@@ -1294,7 +1294,17 @@ void ArxGame::updateFirstPersonCamera() {
 				g_playerCamera.m_pos.x = player.pos.x + vect.x;
 				g_playerCamera.m_pos.z = player.pos.z + vect.z;
 			}
-			
+
+			// ----------------
+
+			int distance = 200.0f;
```
Comment (hogsy) [inline src/core/ArxGame.cpp:1300]: Curious why you're declaring distance as an int. Was this perhaps a mistake?

Comment (meszaros-lajos-gyorgy) [reply src/core/ArxGame.cpp:1300]: good catch, it started out as an int, but later I've changed it to a float. will update it later.

**Comment by meszaros-lajos-gyorgy** [comment]
If anyone stumbles upon this PR then the code have been improved a bit and can be found here: https://github.com/arx-insanity/ArxLibertatis/pull/7/files

**Pull request closed without merging**